libs/titanc/src/compiler/parser.rs

```diff
@@ -10,44 +12,75 @@ use std::str;
 // get the text of a node, we can just get the byte range of the node and then convert
 // that to a string using the source code byte array
 pub struct Parser<'a> {
-  pub source_code: &'a [u8],
+	pub source_code: &'a [u8],
 }
 
 impl<'b> Parser<'b> {
-  pub fn new(source_code: &'b [u8]) -> Self {
-    Self {
-      source_code
-    }
-  }
-
-  pub fn parse(&self, root: &Node) -> Scope {
-    match root.kind() {
-      "source_file" => self.build_scope(&root),
-      _ => panic!("Unexpected root node kind: {}", root.kind()) // tree-sitter parse tree should always start  with source
-    }
-  }
-
-  pub fn build_scope(&self, root: &Node) -> Scope {
-    let span = self.node_span(&root);
-    let node_text = self.node_text(&root);
-    
-    Scope {
-      text: node_text.to_string(),
-      span
-    }
-  }
-
-  // helper function that takes a node and creates a Span from it
-  pub fn node_span(&self, node: &Node) -> Span {
-    let node_range = node.range();
-    Span {
-      start: node_range.start_byte,
-      end: node_range.end_byte,
-    }
-  }
-
-  // helper that gets the text out of a node
-  fn node_text<'a>(&'a self, node: &Node) -> &'a str {
+	pub fn new(source_code: &'b [u8]) -> Self {
+		Self { source_code }
+	}
+
+	pub fn parse(&self, root: &Node) -> Scope {
+		match root.kind() {
+			"source_file" => self.build_scope(&root),
+			_ => panic!("Unexpected root node kind: {}", root.kind()), // tree-sitter parse tree should always start  with source
+		}
+	}
+
+	pub fn build_scope(&self, root: &Node) -> Scope {
+		let span = self.node_span(&root);
+
+		let mut cursor = root.walk();
+
+		let scope = Scope {
+			statements: root
+				.named_children(&mut cursor)
+				.enumerate()
+				.filter_map(|(_, statement_node)| self.build_statement(&statement_node))
+				.collect(),
+			span,
+		};
+
+		scope
+	}
+
+	// helper function that takes a node and creates a Span from it
+	pub fn node_span(&self, node: &Node) -> Span {
+		let node_range = node.range();
+		Span {
+			start: node_range.start_byte,
+			end: node_range.end_byte,
+		}
+	}
+
+	fn build_statement(&self, statement_node: &Node) -> Option<Statement> {
+		let span = self.node_span(statement_node);
+		match statement_node.kind() {
+			"variable_definition" => {
+				let kind = self.build_definition_statement(&statement_node);
+				Some(Statement { kind, span })
+			}
+			_ => None,
```

Comment:
Since our parser should always know every statement kind we add to grammar
```suggestion
			_ => panic!("Unexpected statement node kind: {}", statement_node.kind()),
```